src/wms/src/test/java/org/geoserver/wms/WMSVectorDimensionValidationTest.java

```diff
@@ -0,0 +1,564 @@
+        assertValidationSuccess(
+                setupStartEndDimension(
+                        getCatalog(), TIME_WITH_START_END, dimension, "startTime", "endTime"),
+                request,
+                dimension,
+                dimensionValue,
+                true);
+    }
+
+    /**
+     * For a 'time' dimension, the generated query should have 'maxFeature' set to 1 and the query
+     * properties should have one entry - 'startTime'
+     *
+     * @throws IOException an unexpected IO exception is thrown
+     */
+    @Test
+    public void timeQueryShouldHaveOneAttributeAndLimitMaxFeaturesToOne() throws IOException {
+        // the dimension and its value
+        String dimension = "time";
+        Date dimensionValue = Date.from(Instant.parse("2012-02-11T10:15:30.00Z"));
+        // create a map request
+        GetMapRequest request = mapRequest(singletonList(dimensionValue), emptyList(), emptyMap());
+        // set up a time dimension
+        setupStartEndDimension(
+                getCatalog(), TIME_WITH_START_END, dimension, "startTime", "endTime");
+        // run validation and assert the content of the query
+        assertQueryContent(TIME_WITH_START_END, request, "startTime");
+    }
+
+    /**
+     * For a 'time' dimension, if a mismatching value is provided in the request the validation
+     * should fail.
+     *
+     * @throws IOException an unexpected IO exception is thrown
+     */
+    @Test
+    public void mismatchingTimeDimensionShouldNotValidate() throws IOException {
+        // the dimension and its value
+        String dimension = "time";
+        Date dimensionValue = Date.from(Instant.parse("2012-02-10T10:15:30.00Z"));
+        // create a map request
+        GetMapRequest request = mapRequest(singletonList(dimensionValue), emptyList(), emptyMap());
+        // disable validation
+        setValidationEnabled(true);
+        // run validation and assert
+        assertValidationSuccess(
+                setupStartEndDimension(
+                        getCatalog(), TIME_WITH_START_END, dimension, "startTime", "endTime"),
+                request,
+                dimension,
+                dimensionValue,
+                false);
+    }
+
+    /**
+     * For a 'time' dimension, if a mismatching value is provided in the request the validation
+     * should succeed, in case validation is disabled in the settings.
+     *
+     * @throws IOException an unexpected IO exception is thrown
+     */
+    @Test
+    public void mismatchingTimeDimensionShouldValidateIfValidationDisabled() throws IOException {
+        // the dimension and its value
+        String dimension = "time";
+        Date dimensionValue = Date.from(Instant.parse("2012-02-10T10:15:30.00Z"));
+        // create a map request
+        GetMapRequest request = mapRequest(singletonList(dimensionValue), emptyList(), emptyMap());
+        // disable validation
+        setValidationEnabled(false);
```

Comment:
This one does what it says, instead.

Reply:
Yeah, I went through all comments again and fixed the ones that were not quite right. Thanks for finding them.